spec/requests/users_controller_spec.rb

```diff
@@ -3688,6 +3688,31 @@ def enabled?
       expect(response.status).to eq(200)
       expect(response.parsed_body["cannot_see"][user1.username]).to eq("muted_topic")
     end
+
+    it "returns group which was not invited to topic" do
+      sign_in(Fabricate(:admin))
+
+      get "/u/is_local_username.json", params: {
+        usernames: [group.name], topic_id: private_topic.id
+      }
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["valid_groups"]).to include(group.name)
+      expect(response.parsed_body["cannot_see_groups"][group.name]).to eq("not_allowed")
+    end
+
+    it "does not return group which was was invited to topic" do
+      sign_in(Fabricate(:admin))
+      TopicAllowedGroup.create!(group: group, topic: private_topic)
+
+      get "/u/is_local_username.json", params: {
+        usernames: [group.name], topic_id: private_topic.id
+      }
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["valid_groups"]).to include(group.name)
+      expect(response.parsed_body["cannot_see_groups"][group.name]).to eq(nil)
```

Comment:
I'm confused with `nil` here because a user that cannot see the group will also pass this assertion. In other words, this assertion will pass when a user cannot see the group and when the user can see the group and the topic is not a private message.

Reply:
The whole response is ambiguous to me, I hope the refactor fixed this issue too.